jujugui/static/gui/src/app/components/profile/model-list/model-list.js

```diff
@@ -0,0 +1,256 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+
+'use strict';
+
+/**
+  Model list React component used to display a list of the users models in
+  their profile.
+*/
+class ProfileModelList extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      loadingModels: false,
+      models: null
+    };
+  }
+
+  componentWillMount() {
+    this._fetchModels(this.props.facadesExist);
```

Comment:
In this case we not only need to know that the controller is connected and logged in, but also that the facades have been returned from Juju. If we don't have that then the RPC calls will fail as 'not implemented' because it won't have matching facades.